### ISSUE TITLE
sqlalchemy.exc.CompileError: VARCHAR requires a length on dialect mysql

### ISSUE THREAD

**Issue by zhanghe06** [agronholm/sqlacodegen#65]
sqlacodegen==2.0

```
Traceback (most recent call last):
  File "/Users/zhanghe/code/news_flask_restful/news_flask_restful.env/bin/sqlacodegen", line 11, in <module>
    sys.exit(main())
  File "/Users/zhanghe/code/news_flask_restful/news_flask_restful.env/lib/python2.7/site-packages/sqlacodegen/main.py", line 51, in main
    args.noinflect, args.noclasses)
  File "/Users/zhanghe/code/news_flask_restful/news_flask_restful.env/lib/python2.7/site-packages/sqlacodegen/codegen.py", line 395, in __init__
    not nojoined)
  File "/Users/zhanghe/code/news_flask_restful/news_flask_restful.env/lib/python2.7/site-packages/sqlacodegen/codegen.py", line 149, in __init__
    super(ModelClass, self).__init__(table)
  File "/Users/zhanghe/code/news_flask_restful/news_flask_restful.env/lib/python2.7/site-packages/sqlacodegen/codegen.py", line 77, in __init__
    column.type = self._get_adapted_type(column.type, column.table.bind)
  File "/Users/zhanghe/code/news_flask_restful/news_flask_restful.env/lib/python2.7/site-packages/sqlacodegen/codegen.py", line 94, in _get_adapted_type
    if new_coltype.compile(bind.dialect) != compiled_type:
  File "/Users/zhanghe/code/news_flask_restful/news_flask_restful.env/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py", line 562, in compile
    return dialect.type_compiler.process(self)
  File "/Users/zhanghe/code/news_flask_restful/news_flask_restful.env/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py", line 293, in process
    return type_._compiler_dispatch(self, **kw)
  File "/Users/zhanghe/code/news_flask_restful/news_flask_restful.env/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py", line 81, in _compiler_dispatch
    return meth(self, **kw)
  File "/Users/zhanghe/code/news_flask_restful/news_flask_restful.env/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py", line 2933, in visit_string
    return self.visit_VARCHAR(type_, **kw)
  File "/Users/zhanghe/code/news_flask_restful/news_flask_restful.env/lib/python2.7/site-packages/sqlalchemy/dialects/mysql/base.py", line 1540, in visit_VARCHAR
    self.dialect.name)
sqlalchemy.exc.CompileError: VARCHAR requires a length on dialect mysql

list assignment index out of range
```

but, sqlacodegen==1.1.6, my script works fine
why? so what exactly happened？

**Comment by fabianoengler** [comment]
I am having this issue with mysql and MEDIUMTEXT and LONGTEXT columns. For TEXT columns it works as expected.

I tried several versions of sqlalchemy (from 0.8 to 1.2) and the error happens with all versions I tried.

**Comment by hassan-cedeno** [comment]
THIS is not the proper fix, but it worked for me and so far no further issues: 
in .\dialects\mysql\base.py
modify: 
```
def visit_VARCHAR(self, type_, **kw):
        if type_.length:
            return self._extend_string(
                type_, {}, "VARCHAR(%d)" % type_.length)
        else:
            raise exc.CompileError(
                "VARCHAR requires a length on dialect %s" %
                self.dialect.name)
```
to: 
```
def visit_VARCHAR(self, type_, **kw):
        if type_.length is None:
            type_.length = 255
        if type_.length:
            return self._extend_string(
                type_, {}, "VARCHAR(%d)" % type_.length)
        else:
            raise exc.CompileError(
                "VARCHAR requires a length on dialect %s" %
                self.dialect.name)
```

So basically, if length is None just assumed is max.

**Comment by Timopheym** [comment]
For me sqlacodegen==1.1.6 works too, but 2.0.0  - does not.

**Comment by tr0ublemaker** [comment]
In my case  i find in my database it was a text filed  but got a ''String(collation='utf8mb4_unicode_ci')'' without lenth .  i think is tha because of a bad support of  sqlalchemy.types.Text

**Comment by agronholm** [comment]
Fixed by #75.
